tools/test-js-with-puppeteer

```diff
@@ -86,6 +87,7 @@ def run_tests(files: Iterable[str], external_host: str) -> None:
                 flush=True,
             )
             ret = subprocess.call(cmd)
+            print(f"\n{current_test_num + 1}/{total_tests} tests completed.\n")
```

Comment:
We could add this to the above `===================== message-basics.ts` header instead, perhaps as `===================== (2/16) message-basics.ts`?